tracing/tests/specific_enabled.rs

```diff
@@ -0,0 +1,38 @@
+// liballoc is required because the test subscriber cannot be constructed
+// statically
+#![cfg(feature = "alloc")]
+
+mod support;
+
+use self::support::*;
+use tracing::Level;
+
+#[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
+#[test]
+fn span_and_event() {
+    let collector = collector::mock()
+        .with_filter(|meta| {
+            if meta.target() == "debug_module" {
+                meta.level() <= &Level::DEBUG
+            } else if meta.is_span() {
+                meta.level() <= &Level::TRACE
+            } else if meta.is_event() {
+                meta.level() <= &Level::DEBUG
+            } else {
+                meta.level() <= &Level::INFO
+            }
+        })
+        .done()
+        .run();
+
+    tracing::collect::set_global_default(collector).unwrap();
```

Comment:
i would probably use the local default (`set_default`) in these tests, so they can all be in the same file...having to put it in a separate file is kind of annoying. not a hard blocker though.

Reply:
rather, ill clean this up, then yall can squash-and-merge